docs/DeveloperGuide.md

```diff
@@ -1,34 +1,442 @@
+- implements methods to add new patients and delete existing patients
+- implements methods to load an existing patient's medical records
+
+A `Patient` contains:
+
+- the patient's NRIC/FIN number, which uniquely identifies the patient
+- a `TreeMap<LocalDate, Record>` which maps the patient's consultation dates to the visit records for that date
+
+A `Record` contains:
+
+- all the symptoms recorded by a GP during the consultation
+- all the diagnoses made by a GP during the consultation
+- all the prescriptions made by a GP during the consultation
+- the most recently added symptom/diagnosis/prescription, which corresponds to the most recently executed `record` command
+
+
+### Storage Component
+
+API: `Storage.java`
+
+The storage component is in charge of printing all patients' data into a file, and loading from it when the application restarts. 
+This would ensure that patient data is not lost after exiting the application.
+
+To initialise this component, the `SortedMap<String, Patient>` from the Data class is passed during object creation.
+This SortedMap<String, Patient> is the entire list of patients and their records.
+
+Then, we can call the `save(SortedMap<String, Patient> patientData)` method to save the SortedMap into a file whenever changes are made.
+The file is specified by the variable FILE_PATH in Constants class.
+
+The reverse process would be the `load()` method. The file from the specified FILE_PATH is loaded into a String, which is 
+converted to a SortedMap<String, Patient> afterwards.
+
+`Storage.java`:
+
+- facilitates the saving of application data into a text file
+- facilitates the loading of application data from the aforementioned text file
+- convert records to string
+- converts string to records
+
+### Exception Component
+
+API: `BaseException.java` and its subclasses
+
+`BaseException.java`:
+- handles all exceptions that occur during the execution of PatientManager
+- can report an error message, prompting the user to provide a syntatically correct command
+- may also report the cause of error for debugging purposes
+
+### Commons
+
+Constants used by multiple classes are stored in the `seedu.duke.Constants` class.
+
+## Implementation
+
+This section describes some noteworthy details on how certain details are implemented.
+
+### Tokenizing User Input
+
+The parser is one of the core components in charge of parsing all user input commands into program-understandable commands and
+arguments. For the ease of expansion of this program's functionality as well as for its testability, reflection is used to invoke
+commands.
+
+First is the initialization of this parser. A `Ui` instance and a `Data` instance is passed and stored. This is important as
+these two will be passed to logic components (command classes) later.
+
+Then, we can parse a user-input string by passing it to `parse()`. We use an example of this:
+```
+record 01/05/2021 /s coughing, fever /p panadol Paracetamol 500mg*20
+```
```

Comment:
May want to include a format for the input string first, before providing an example.
For instance: ```record (dd/MM/YY) /s (symptom), (sickness) /p (medicine dosage)```
A further explanation on the key words like ```record```, ```/s``` and ```/p``` would be nice too